version: '3.8'

services:
  minio:
    image: minio/minio:latest
    container_name: minio
    command: server /data --console-address ":9001"
    ports:
      - "9000:9000"
      - "9001:9001"
    environment:
      MINIO_ROOT_USER: engineer
      MINIO_ROOT_PASSWORD: MinioSecurePass123!
    volumes:
      - minio_data:/data
    restart: unless-stopped

  postgres:
    image: postgres:15-alpine
    container_name: postgres
    environment:
      POSTGRES_USER: dwadmin
      POSTGRES_PASSWORD: PgSecurePass123!
      POSTGRES_DB: datawarehouse
    ports:
      - "5432:5432"
    volumes:
      - postgres_data:/var/lib/postgresql/data
    restart: unless-stopped

  jupyter:
    image: jupyter/datascience-notebook:notebook-6.5.4
    container_name: jupyter
    user: root
    ports:
      - "8888:8888"
    volumes:
      - ./notebooks:/home/jovyan/work
      - ./scripts:/home/jovyan/scripts
    environment:
      JUPYTER_TOKEN: "DataEngChallenge2025!"
      NB_USER: jovyan
      NB_UID: ${UID}
      NB_GID: ${GID}
      CHOWN_HOME: yes
    working_dir: /home/jovyan
    command: 
      - start-notebook.sh
      - --ip=0.0.0.0
      - --NotebookApp.allow_origin=*
      - --NotebookApp.allow_remote_access=1
    restart: unless-stopped

volumes:
  minio_data:
  postgres_data: